File: queues/nats/C/tests/test_natsclient.c
#include "../../../../tests/c/testutil.h"
#include "../include/natsclient.h"
#include <stdlib.h>

TEST(test_init_missing)
{
    unsetenv_safe("NATS_URL");
    unsetenv_safe("NATS_SERVERS");
    char *json = nats_init();
    ASSERT("601", json_status(json) == 601);
    free(json);
}

TEST(test_init_ok)
{
    setenv("NATS_URL", "nats://localhost:4222", 1);
    char *json = nats_init();
    ASSERT("200", json_status(json) == 200);
    free(json);
}

int main(void)
{
    RUN(test_init_missing);
    RUN(test_init_ok);
    printf("nats C tests passed\n");
    return 0;
}

File: queues/nats/Python/tests/test_natsclient.py
import sys
from pathlib import Path
from unittest.mock import AsyncMock, patch

import pytest

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
import natsclient


@pytest.fixture(autouse=True)
def env(monkeypatch):
    monkeypatch.setenv("NATS_URL", "nats://localhost:4222")


def test_init_missing(monkeypatch):
    monkeypatch.delenv("NATS_URL", raising=False)
    monkeypatch.delenv("NATS_SERVERS", raising=False)
    assert natsclient.Init()["status_code"] == 601


@patch("natsclient.asyncio.run")
def test_publish(mock_run):
    mock_run.return_value = {"status_code": 200}
    assert natsclient.Publish("subj", {"a": 1}) == {"status_code": 200}
